www/src/components/layout.js

```diff
@@ -20,7 +20,7 @@ export default function DefaultLayout({ location, locale, children }) {
   return (
     <LocaleContext.Provider value={locale || defaultLang}>
       <Global styles={globalStyles} />
-      <SiteMetadata pathname={location.pathname} locale={locale} />
+      <SiteMetadata href={location.href} locale={locale} />
```

Comment:
when the above link is the current status link to the build system, 

then `location.href` is not available at SSR:

Check:
```bash
curl https://build-5fdd9a37-e2ae-4a55-a681-3bea24c30edb.gtsb.io/showcase/reactjs.org | grep "canonical" | less
```
results in:
```html
<link data-react-helmet="true" rel="canonical"/>
<meta data-react-helmet="true" name="docsearch:version" content="2.0"/>
<meta data-react-helmet="true" name="viewport" content="width=device-width,initial-scale=1,shrink-to-fit=no,viewport-fit=cover"/>
<meta data-react-helmet="true" property="og:url"/>
```

while 
```bash
curl https://www.gatsbyjs.org/showcase/reactjs.org | grep "canonical" | less
```

```html
<link data-react-helmet="true" rel="canonical" href="https://www.gatsbyjs.org/showcase/reactjs.org"/>
<meta data-react-helmet="true" name="docsearch:version" content="2.0"/>
<meta data-react-helmet="true" name="viewport" content="width=device-width,initial-scale=1,shrink-to-fit=no,viewport-fit=cover"/>
<meta data-react-helmet="true" property="og:url" content="https://www.gatsbyjs.org"/>
```

Reply:
Thanks for the tip! I'll change it back to using pathname. I didn't know that.

Reply:
but check it before with a current build

i remember there was that problem, but unfortunly i can not find some issues or some docs ([troubleshooting](https://www.gatsbyjs.org/docs/troubleshooting-common-errors/)) about this ..

Reply:
added issue:
- `docs: add info about location.href` 

thanks for the link to the blog post in code comments :)